### ISSUE TITLE
Webhook on `UP`/`DOWN` events

### ISSUE THREAD

**Issue by xoraingroup** [louislam/uptime-kuma#942]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

With the API call being defined with all headers and payload in UI and on any **UP** or **DOWN** event, this will be very useful in case if user wants to notify a remote system with custom data. This will also make it very easy to integrate it basically with any remote system that supports incoming API calls.

### ✔️ Solution

1. Should be possible to use GET, POST, PATCH method at least
2. Should be possible to add custom headers with their respective values. This will solve problem of any authorization based on tokens or api keys and also possibility of more custom headers.
3. Should be possible to add payload as application/x-www-form-urlencoded or application/json.
4. Possibility to pass query params

**Comment by CommanderStorm** [comment]
So basically, you would like webhooks?
![image](https://github.com/louislam/uptime-kuma/assets/26258709/cb99427b-4db1-4c07-9058-bf29bcd9d95e)

Note that there are a lot of issues out there which explain this more clearly like:
- HTTP-Methods: https://github.com/louislam/uptime-kuma/issues/2413
- custom body will be released in `1.23` as https://github.com/louislam/uptime-kuma/pull/3088 was merged
- variables in query strings are quite uncommon => not a lot of users are requesting these: https://github.com/louislam/uptime-kuma/issues/2813

**Comment by CommanderStorm** [comment]
TLDR: 
could you edit the feature request to only include query strings?

**Comment by CommanderStorm** [comment]
Closing as resolved by webhooks or better tracked in https://github.com/louislam/uptime-kuma/issues/508 https://github.com/louislam/uptime-kuma/issues/2413 as stated above
